extensions/http-curl/processors/InvokeHTTP.cpp

```diff
@@ -65,7 +65,13 @@ core::Property InvokeHTTP::FollowRedirects("Follow Redirects", "Follow HTTP redi
 core::Property InvokeHTTP::AttributesToSend("Attributes to Send", "Regular expression that defines which attributes to send as HTTP"
                                             " headers in the request. If not defined, no attributes are sent as headers.",
                                             "");
-core::Property InvokeHTTP::SSLContext("SSL Context Service", "The SSL Context Service used to provide client certificate information for TLS/SSL (https) connections.", "");
+core::Property InvokeHTTP::SSLContext("SSL Context Service",
+                                      "The SSL Context Service used to provide client certificate "
+                                      "information for TLS/SSL (https) connections.",
+                                      "",
+                                      false,
+                                      {},
+                                      {{"Remote URL", "^http:.*$"}});
```

Comment:
shouldn't this be https? I also think that URL regexes tend to be much more complicated than .*

Reply:
Actually, is this option part of SSL Context Service? Seems like the wrong variable name for the property.

Reply:
@apiri mentioned his interpretation of this. let's change this and use a builder instead. This is difficult to read.

Reply:
would be in favor of that

Reply:
This is confusing because the PR is based on MINIFICPP-500 mutually-exclusive. This PR is for MINIFICPP-514 so that Remote URL mutually-exclusive property is correct.

A URL regex could get complex, but in this case we're just checking for HTTP protocol as per RFC3986 (which is exclusive of this SSL property, for mutual-exclusivity as defined by MINIFICPP-467 E.g. "If I set certificates in InvokeHTTP, I shouldn’t allow http as the target protocol.").

Should it use a controller service? Probably. That's out of scope for this ticket but I agree with a new ticket for that.

All these notes apply for MINIFICPP-500, not MINIFICPP-514.

Reply:
To clarify, the regex is just checking for positive incidence of HTTP protocol, nothing more. It isn't there to validate a URL (again this is very confusing due to the incidental git basing).

Reply:
I'm afraid I'm going to disagree. I'm -1 on this PR until that issue is addressed. Completely happy with it being a separate ticket.

Reply:
I see this comment chain can be a bit confusing -- there were offline discussions that make it a bit worse. My request for a separate ticket was that this and other PRs have made the usage of these classes very difficult to read and use. I believe that it's bad enough that it warrants resolution in this ticket or subsequent work. To make code worse and then fix it doesn't seem correct; however, I'm okay with a subsequent ticket. I'm -1 with my opinion merit especially since there was no discussion of this. I think Aldrin and others can disagree and move forward with their +1s, through which I'll provide an implicit agreement to move forward -- I'm also open to the possibility that I'm completely wrong so let's actually have that discussion. My goal is not to stop this but there was no discussion regarding how this and others have made the code difficult to read and use.